pkg/injector/patch.go

```diff
@@ -38,24 +37,71 @@ func (wh *mutatingWebhook) createPatch(pod *corev1.Pod, req *admissionv1.Admissi
 	metricsstore.DefaultMetricsStore.CertIssuedCount.Inc()
 	metricsstore.DefaultMetricsStore.CertIssuedTime.
 		WithLabelValues().Observe(elapsed.Seconds())
-	originalHealthProbes := rewriteHealthProbes(pod)
+	originalHealthProbes := rewriteHealthProbes(pod) // need to get these out of here...
 
 	// Create the bootstrap configuration for the Envoy proxy for the given pod
-	envoyBootstrapConfigName := fmt.Sprintf("envoy-bootstrap-config-%s", proxyUUID)
-
-	// The webhook has a side effect (making out-of-band changes) of creating a k8s
-	// Secret corresponding to the Envoy bootstrap config. Such a side effect needs to be
-	// skipped when the request is a DryRun.
+	envoyBootstrapConfigName := bootstrapSecretPrefix + proxyUUID.String()
+
+	// Pods definitions can be copied via the `kubectl debug` command, which can lead to a pod being created that
+	// has already had injection occur. We could simply do nothing and return early, but that would leave 2 pods
+	// with the same UUID, so instead we change just the UUID, and its associated bits.
+	// This needs to occur before replacing the label below.
+	originalUUID, alreadyInjected := getUUID(pod)
+	// The webhook has a side effect (making out-of-band changes) of creating k8s secret
+	// corresponding to the Envoy bootstrap config. Such a side effect needs to be skipped
+	// when the request is a DryRun.
```

Comment:
I'd move this comment just above L55, next to its case in the switch statement.
Same applies to `case alreadyInjected`.

Reply:
done!